motor/render/vulkan_renderer.cpp

```diff
@@ -205,33 +243,59 @@ vk::SwapchainKHR CreateSwapChain(const vk::PhysicalDevice& phy_dev,
       VkSuccuessOrDie(phy_dev.getSurfacePresentModesKHR(vk_surface),
                       "Couldn't get present modes");
 
-  vk::Extent2D swap_chain_extend;
+  vk::Extent2D swap_chain_extent;
   // TODO(kadircet): Use values coming from user instead.
-  if (surf_caps.currentExtent.height == 0xFFFFFFFF)
+  if (surf_caps.currentExtent.height == UINT32_MAX)
   {
     LOG(INFO) << "Current extent unknown setting to 800x600";
-    surf_caps.currentExtent.setHeight(600);
-    surf_caps.currentExtent.setWidth(800);
+    uint32_t width = 800;
+    uint32_t height = 600;
+    surf_caps.currentExtent.setWidth(
+        std::max(surf_caps.minImageExtent.width,
+                 std::min(surf_caps.maxImageExtent.width, width)));
+    surf_caps.currentExtent.setHeight(
+        std::max(surf_caps.minImageExtent.height,
+                 std::min(surf_caps.maxImageExtent.height, height)));
+  }
+  swap_chain_extent = surf_caps.currentExtent;
+
+  uint32_t image_count = surf_caps.minImageCount + 1;
+  if (surf_caps.maxImageCount > 0 && image_count > surf_caps.maxImageCount)
```

Comment:
what does it mean to have `maxImageCount == 0` ? if it is an error can we just report it and bail out, and then just use `image_count = std::min(maxImageCount, minImageCount + 1)`.

Also why do we wan't `minImageCount + 1`? Do we expect it to be 2? If so it might be better to say it out explicitly.